lib/cretonne/src/legalizer/mod.rs

```diff
@@ -92,6 +92,8 @@ pub fn legalize_function(func: &mut Function, cfg: &mut ControlFlowGraph, isa: &
                     if changed {
                         pos.set_position(prev_pos);
                         continue;
+                    } else {
+                        func.encodings.ensure(inst);
```

Comment:
This is a good invariant to enforce. I wonder if it is simpler and safer to call `encodings.resize(dfg.num_insts())` after the legalizer pass, though?